src/typescript-service.ts

```diff
@@ -32,11 +34,30 @@ export default class TypeScriptService {
         this.root = root;
         this.projectManager = new pm.ProjectManager(root, strict, connection);
 
+        this.fetchDefinitelyTypedRepo();
+
         //initialize providers 
         this.exportedSymbolProvider = new ExportedSymbolsProvider(this);
         this.externalRefsProvider = new ExternalRefsProvider(this);
     }
 
+    /**
+     * Fetch repository from https://github.com/DefinitelyTyped/DefinitelyTyped into /tmp/DefinitelyTyped/
+     */
+    fetchDefinitelyTypedRepo() {
+        fs.exists('/tmp/DefinitelyTyped', (exists) => {
```

Comment:
We don't need to fetch node_modules anymore and we actually are not fetching them. All we need is to download DefinitelyTyped repo somewhere, this repo will be shared for all processes (claster mode), all users etc. And update it periodically since people updating d.ts files (once a day is enough) or while restarting of sourcegraph. Size of repo is not so big 128 Mb. 
Maybe it would be easier if I explain it this way: imagine that these d.ts files are source code files of langserver project located in third part repo - DefinitelyTyped/DefinitelyTyped repo.